packages/vite/src/utils.ts

```diff
@@ -16,8 +15,7 @@ export async function generateImageID(url: URL, config: ImageConfig, imageBuffer
   // baseURL isn't a valid URL, but just a string used for an identifier
   // use a relative path in the local case so that it's consistent across machines
   const baseURL = new URL(url.protocol + path.relative(process.cwd(), url.pathname))
-  const { mtime } = statSync(path.resolve(process.cwd(), decodeURIComponent(url.pathname)))
-  return hash([baseURL.href, JSON.stringify(config), mtime.getTime().toString()])
+  return hash([baseURL.href, JSON.stringify(config), imageBuffer])
```

Comment:
hashing the image buffer has to be the most expensive part of this since it's so large relative to everything else. this has us re-calculating a hash for the image for each individual image config, which we probably don't need to do. what if we calculate the hash for the image separately first and then make the third parameter of this function be the hash of the image rather than the image itself